File: ContactsManagerAppWithJavaROOMDBMVVM/app/src/main/java/com/masterclass/contactsmanagerappwithjavaroomdbmvvm/activityClickHandlers/MainActivityClickHandlers.java
package com.masterclass.contactsmanagerappwithjavaroomdbmvvm.activityClickHandlers;

import android.content.Context;
import android.content.Intent;
import android.view.View;

import com.masterclass.contactsmanagerappwithjavaroomdbmvvm.activities.AddNewContactActivity;

public class MainActivityClickHandlers {
    Context context;

    public MainActivityClickHandlers(Context context) {
        this.context = context;
    }

    public void onFABClick(View view){
        Intent i = new Intent(context.getApplicationContext(), AddNewContactActivity.class);
        context.startActivity(i);
    }
}

File: ContactsManagerAppWithKotlinROOMDBMVVM2/app/src/main/java/com/masterclass/contactsmanagerappwithkotlinroomdbmvvm/activityClickHandlers/MainActivityClickHandlers.kt
package com.masterclass.contactsmanagerappwithkotlinroomdbmvvm.activityClickHandlers

import android.content.Context
import android.content.Intent
import android.view.View
import com.masterclass.contactsmanagerappwithkotlinroomdbmvvm.activities.AddNewContactActivity

class MainActivityClickHandlers(private val context: Context) {

    fun onFABClick(view: View) {
        val intent = Intent(context.applicationContext, AddNewContactActivity::class.java)
        context.startActivity(intent)
    }
}